internal/searcher/client.go

```diff
@@ -32,31 +32,40 @@ var cachedClient *client
 func Client() *client {
 	if cachedClient == nil {
 		endpoint, _ := url.JoinPath(searchHost(), "search")
+		// TODO: how to handle error
 		cachedClient = &client{
 			endpoint: endpoint,
 		}
 	}
 	return cachedClient
 }
 
-func (c *client) Search(query string) (*SearchResult, error) {
-	return execSearch(c.endpoint + "?q=" + url.QueryEscape(query))
+func (c *client) Search(query string, options ...func() string) (*SearchResult, error) {
+	if query == "" {
+		return nil, fmt.Errorf("query should not be empty")
+	}
+
+	searchURL := c.endpoint + "?q=" + url.QueryEscape(query)
+
+	for _, op := range options {
+		searchURL += op()
```

Comment:
Another possible signature for the option function is a function that takes the endpoint, modifies it and returns the new one. But this is OK as well. If we ever add more options we can generalize if needed

Reply:
> Another possible signature for the option function is a function that takes the endpoint, modifies it and returns the new one. But this is OK as well. If we ever add more options we can generalize if needed

Yeah, I first came up with this one. But here's a problem: each option needs to handle some similar cases for 0 or more queries. So I finally decided to return the query string in the option to  be appended to the endpoint.

If we have more options, we can see which will be better.